### PR TITLE
ERC20 support

### PR DESCRIPTION
**Pull request by rajranjan0608**
With this PR, we can add ERC20 faucets as well. See below a sample configuration, for adding Wrapped WAGMI token from the WAGMI subnet (HOSTID)

```json
"erc20tokens": [
        {
            "ID": "WWGMI",
            "HOSTID": "WAGMI",
            "NAME": "Wrapped WAGMI",
            "TOKEN": "WWGMI",
            "CONTRACTADDRESS": "0x3Ee7094DADda15810F191DD6AcF7E4FFa37571e4",
            "DRIP_AMOUNT": 1000000000,
            "GASLIMIT": "150000",
            "DECIMALS": 18,
            "RATELIMIT": {
                "MAX_LIMIT": 2,
                "WINDOW_SIZE": 1440
            }
        }
    ]
```
It will derive the rest of the information like RPC, MAX_FEE, etc from the HOSTID

### CODE REVIEW

File: client/src/components/AddNetwork.tsx [hunk 1]
```diff
@@ -32,6 +32,30 @@ const addNetwork = async (config: any) => {
...
 
+const addAsset = async (config: any) => {
+    if(!config) {
+        return;
+    }
+    if(window.ethereum == undefined) {
+        window.open('https://metamask.io/download', '_blank');
+    }
+
+    console.log(config)
```
Comment (cgcardona) [inline client/src/components/AddNetwork.tsx]: Can you remove all `console.log` statements

File: client/src/components/FaucetForm.tsx [hunk 2]
```diff
@@ -49,56 +51,117 @@ const FaucetForm = (props: any) => {
...
+    1&2 - erc20 tokens of only subnet id in the list
+    1&3 - effect of 3
+    2&3 - effect of 3
+    */
+
     useEffect(() => {
         const query = queryString.parse(window.location.search);
-        if(typeof query?.address == "string") {
+        
+        const { address, type, subnet, erc20 } = query;
```
Comment (cgcardona) [inline client/src/components/FaucetForm.tsx]: Can you remove all semicolons?

File: client/src/components/FaucetForm.tsx [hunk 3]
```diff
@@ -49,56 +51,117 @@ const FaucetForm = (props: any) => {
...
+            params.chain = chainConfigs[chain!]?.ID;
+        }
+        return params;
     }
 
     async function updateBalance() {
         if((chain || chain == 0) && chainConfigs.length > 0) {
-            const response = await props.axios.get(props.config.api.getBalance, {params: {chain: chainConfigs[chain!]?.ID}});
+            let { chain, erc20 } = getChainParams()
+            const response = await props.axios.get(props.config.api.getBalance, { params: {chain, erc20} });
```
Comment (cgcardona) [inline client/src/components/FaucetForm.tsx]: Please add types to all `let` and `const` statements.

File: client/src/components/FaucetForm.tsx [hunk 4]
```diff
@@ -176,6 +242,16 @@ const FaucetForm = (props: any) => {
         setIsLoading(false);
     }
 
+    const getOptionByValue = (value: any) => {
```
Comment (cgcardona) [inline client/src/components/FaucetForm.tsx]: Please add return value to functions

File: client/src/components/FaucetForm.tsx [hunk 5]
```diff
@@ -176,6 +242,16 @@ const FaucetForm = (props: any) => {
...
+        let selectedOption: any = options[0];
+        options.forEach((option: any) => {
```
Comment (cgcardona) [inline client/src/components/FaucetForm.tsx]: Please add return values to all `forEach` statements. If there is no return value then use `void`.

File: server.ts [hunk 6]
```diff
@@ -22,12 +22,35 @@ app.use(cors())
...
+    let reply;
+    chains.forEach((chain: any) => {
+        if(chain.ID == id) {
+            reply = chain;
+        }
+    });
+    return reply;
+}
+
+const populateConfig = (child: any, parent: any) => {
```
Comment (cgcardona) [inline server.ts]: I see `any` as a bunch of types. Do we not know the actual type? VSCode will often tell you if you hover your mouse over the variable for function.

Comment (rajranjan0608) [reply server.ts]: I have updated the code with new types wherever possible. Thanks.

**Pull request merged**